From e4af35fdfcd6115741982fe0371d1b6b5f8c6db4 Mon Sep 17 00:00:00 2001
From: Richard Yao <ryao@gentoo.org>
Date: Mon, 5 Nov 2012 13:32:08 -0500
Subject: [PATCH] Make missing ZFS config file failures into warnings, bug
 #438200

Whenever /etc/zfs/{zdev.conf,zpool.cache} was missing, we would fail.
zpool.cache is not generated during cross compilation, which causes a
failure in Catalyst when building LiveCDs. Users that do not follow a
strict set of installation instructions can also suffer failures as a
consequence of this.

zpool.cache includes information about known pools that enables ZFS to
detect situations where an entire pool disappears. It is also required
to do pool import in corner cases, such as those those involving
file-based pools, and has the beneficial effect of reducing pool import
times.  Unconditional omission of zpool.cache in general is not an
option, but for the situations identified, it is okay to convert the
failure into a warning, which is what we do.

Reported-by: Rick Farina <zero_chaos@gentoo.org>
Reported-by: Alexander Zubkov <green@msu.ru>
Signed-off-by: Richard Yao <ryao@gentoo.org>
---
 gen_initramfs.sh |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gen_initramfs.sh b/gen_initramfs.sh
index 64e00e8..99bd3f3 100755
--- a/gen_initramfs.sh
+++ b/gen_initramfs.sh
@@ -410,7 +410,7 @@ append_zfs(){
 	for i in /etc/zfs/{zdev.conf,zpool.cache}
 	do
 		cp -a "${i}" "${TEMP}/initramfs-zfs-temp/etc/zfs" \
-			|| gen_die "Could not copy file ${i} for ZFS"
+			|| print_warning 1 "Could not copy file ${i} for ZFS"
 	done
 
 	# Copy binaries
-- 
1.7.8.5

